### DISCUSSION TITLE
Processing Time estimation

### DISCUSSION THREAD

**Discussion by montagao** [SYSTRAN/faster-whisper#233] [Q&A]
Has anyone figured out how to get a relatively accurate processing time estimate for a given audio file and given a fixed set of hardware specifications? 

Some formula involving the file size?

Does it vary greatly between language? 

What would be a good approach?

**Comment by guillaumekln** [comment]
The only approach is to actually run the program on the target hardware and measure the time for different audio files. Then you can interpolate on other audio duration.

Just note that by default the transcription time for a single audio file can vary unpredictably. You can search this repo or openai/whisper about the "temperature fallback" where the model will sample transcriptions randomly. In that case the transcription time can vary from one run to another.

**Reply by montagao** [reply]
Ah okay. Thanks for clearing that up.

**Comment by phineas-pta** [comment]
you can add a `tqdm` progress bar to get estimated time see #80
